### DISCUSSION TITLE
Line: hard to vary serie by anything but color

### DISCUSSION THREAD

**Discussion by jamesarosen** [plouc/nivo#2512] [Q&A]
The way Nivo's APIs for Line charts work, each series varies only by color. All the series in a single chart are either solid or dashed and all have the same SymbolShape. For folks who are unable to see certain distinctions in color, it's helpful to add additional differences to make the series stand out.

It would be nice to be able to configure more properties by series. Ideally, I'd like to be able to group them together in the theme:

```ts
series: [
  { color: 'purple', strokeWidth: 2, symbolShape: 'square' },
  { color: 'orange', strokeWidth: 2, strokeDasharray: '3, 3', symbolShape: 'diamond' },
  { color: 'green', strokeWidth: 3, strokeDasharray: '4, 2', pointSize: 6, symbolShape: 'circle' },
]
```

Even better would be if I could easily mix and match areas and lines for further contrast!

Here's an example that took about 15 minutes in [AG Charts](https://charts.ag-grid.com/):

![image](https://github.com/plouc/nivo/assets/2406/9ae4418d-9c92-4b81-9309-6cae98fc08a6)

It's _possible_ to do some of this with custom layers, but it's definitely working against the framework.

**Comment by jamesarosen** [comment]
Chart.js has an [excellent example](https://www.chartjs.org/docs/latest/samples/line/styling.html) of this:

```ts
const data = {
  labels: labels,
  datasets: [
    {
      label: 'Unfilled',
      fill: false,
      backgroundColor: Utils.CHART_COLORS.blue,
      borderColor: Utils.CHART_COLORS.blue,
      data: Utils.numbers(NUMBER_CFG),
    }, {
      label: 'Dashed',
      fill: false,
      backgroundColor: Utils.CHART_COLORS.green,
      borderColor: Utils.CHART_COLORS.green,
      borderDash: [5, 5],
      data: Utils.numbers(NUMBER_CFG),
    }, {
      label: 'Filled',
      backgroundColor: Utils.CHART_COLORS.red,
      borderColor: Utils.CHART_COLORS.red,
      data: Utils.numbers(NUMBER_CFG),
      fill: true,
    }
  ]
};
```

**Comment by SourceCipher** [comment]
Yes we are missing this customization for these components as well. For example the prop `pointColor` can be rendered as func and they passing the prop to the function - whole data.. Why not pass circles index as well so we could color each circle separately??? Very strange..
